### PR TITLE
Automate the release process

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

- Only support Druid 26 & 27
- Move the version name to coincide with the SDP release
- Activate LTS version by default
- Add a Github workflow that does a Maven and GitHub release

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
```

```[tasklist]
# Acceptance
- [ ] Proper release label has been added
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+name: Build and Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-version:
+        description: 'Version (and git tag) that this release should use'
```
Comment (sbernauer) [inline .github/workflows/release.yml:8]: Can we please have an `e.g. 1.2.3` example on what is expected here? `23.7.0`?

Can we please also clean up the tags in https://github.com/stackabletech/druid-opa-authorizer/tags (e.g. after merging this PR)?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        required: true
+      development-version:
+        description: 'Version of the next development version WITHOUT the "-SNAPSHOT" at the end'
```
Comment (sbernauer) [inline .github/workflows/release.yml:11]: Can we please have an `e.g. 1.2.3` example on what is expected here?

File: pom.xml [hunk 3]
```diff
@@ -293,45 +310,14 @@
   For the last years the API interface to Druid didn't change and while the versions of dependencies changed and even the location of classes in JARs the actual interface did not.
   In a compiled Java class the bytecode refers to a String constant of the class it needs to access (e.g. org/apache/druid/server/security/Action) and as long as that name (and its interface) does not change it does not matter in which JAR file the code lives.
   Using these profiles we can get clean compiles against all supported versions but we only use those for CI.
-  The actual final compilation will only happen against a single version and for that we chose the latest Druid version we support.
+  The actual final compilation will only happen against a single version and for that we chose the LTS version of Druid for this particular SDP release.
```
Comment (sbernauer) [inline pom.xml:313]: Well that's interesting... :thinking:

File: pom.xml [hunk 4]
```diff
@@ -293,45 +310,14 @@
...
-          <version>${druid.version}</version>
-          <scope>provided</scope>
-        </dependency>
-      </dependencies>
-    </profile>
     <profile>
       <id>druid-26</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
```
Comment (sbernauer) [inline pom.xml:320]:
```suggestion
      <!-- LTS version as of 23.11 -->
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
```

File: pom.xml [hunk 5]
```diff
@@ -4,7 +4,8 @@
 
   <groupId>tech.stackable</groupId>
   <artifactId>druid-opa-authorizer</artifactId>
-  <version>0.6.0-SNAPSHOT</version>
+  <version>23.11-SNAPSHOT</version>
```
Comment (sbernauer) [inline pom.xml:7]: Must it be `23.11-SNAPSHOT` or `23.11.0-SNAPSHOT`? Omitting the patch level will prohibit us from fixing bugs and releasing them as 23.11.1 I fear.

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (lfrancke) [reply pom.xml:313]: What exactly? Anything you want me to change?

File: pom.xml (hunk 5 above)
Comment (lfrancke) [reply pom.xml:7]: Good point. Will add a comment to the PR to discuss further.

**Comment by lfrancke** [comment]
Thanks for the review.

My idea was to basically have one version of druid-opa-authorizer per SDP release but I didn't think it through entirely.
We do not necessarily want to release a new version for every patch release of SDP which means we'd have to pass in the opa authorizer version as a variable in the docker images. And if we do that...how much benefit is it of having these versions?
Will it create more confusion?

We can also stay with 0.6, 0.7 etc.
Opinions?

**Comment by lfrancke** [comment]
btw. my tests including releases can be seen here: https://github.com/lfrancke/druid-opa-authorizer/releases

File: pom.xml (hunk 3 above)
Comment (sbernauer) [reply pom.xml:313]: Nope, I just was suprised

**Comment by sbernauer** [comment]
I haven't fully though this through. But sticking to 0.5.0 seems like an option to me. We could bump the version in case

1.) We switch the Druid LTS line
2.) We fix a bug in druid-opa-authorizer
3. New OPA version with breaking change supported (pretty unlikely)

So - besides security updates - a release approx. ever 4 releases seems sufficient to me

**Comment by lfrancke** [comment]
Could you ask for opinions in the arch meeting?
I do not have a strong opinion and won't be there this week.
Thank you

**Comment by sbernauer** [comment]
Sure, added an entry

**Comment by lfrancke** [comment]
This is ready for review again.

File: pom.xml [hunk 6]
```diff
@@ -293,45 +309,15 @@
   For the last years the API interface to Druid didn't change and while the versions of dependencies changed and even the location of classes in JARs the actual interface did not.
   In a compiled Java class the bytecode refers to a String constant of the class it needs to access (e.g. org/apache/druid/server/security/Action) and as long as that name (and its interface) does not change it does not matter in which JAR file the code lives.
   Using these profiles we can get clean compiles against all supported versions but we only use those for CI.
-  The actual final compilation will only happen against a single version and for that we chose the latest Druid version we support.
+  The actual final compilation will only happen against a single version and for that we chose the LTS version of Druid for this particular SDP release.
```
Comment (sbernauer) [inline pom.xml:312]: I think that's outdated now
```suggestion
```

**Review by sbernauer — COMMENTED** [review]
LGTM otherwise

**Review by sbernauer — APPROVED** [review]

**Pull request merged**